pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,376 @@
+                        this.setState({ showModal: false, error: "" });
+                    });
+                });
+            })
+                    .catch(e => this.setState({ error: e.toString() }))
+                    .finally(() => {
+                        this.checkSetup();
+                        this.setState({ updating: false });
+                    });
+        }
+    }
+
+    render() {
+        // Not yet recognized
+        if (this.state.loaded === false || this.state.patchName === null)
+            return null;
+
+        // Not supported on this system
+        if (this.state.unavailable.length > 0)
+            return null;
+
+        let state = _("Enabled");
+        let actionText = _("Edit");
+        let action = () => this.setState({ showModal: true });
+
+        if (this.state.missing.length > 0 && this.state.unavailable.length === 0) {
+            state = _("Not installed");
+            actionText = _("Install");
+            action = this.handleInstall;
+        } else if (!this.state.enabled) {
+            state = _("Disabled");
+            actionText = _("Enable");
+        }
+
+        let kernel_name = "";
+        if (this.state.kernelName)
+            kernel_name = " (" + this.state.kernelName + ")";
+
+        let error = null;
+        if (this.state.error)
+            error = <Alert variant='danger' isInline title={this.state.error} />;
+
+        const body = <Checkbox id="apply-kpatch"
+                               isChecked={this.state.applyCheckbox}
+                               label={_("Apply kernel patches")}
+                               onChange={checked => this.setState({ applyCheckbox: checked })}
+                               body={<>
+                                   <Radio id="current-future" label={_("for current and future kernels")} onChange={() => this.setState({ justCurrent: false })} isDisabled={!this.state.applyCheckbox} isChecked={!this.state.justCurrent} />
+                                   <Radio id="current-only" label={_("for current kernel only") + kernel_name} onChange={() => this.setState({ justCurrent: true })} isDisabled={!this.state.applyCheckbox} isChecked={this.state.justCurrent} />
+                               </>}
+        />;
+
+        return (<>
+            <div id="kpatch-settings">
+                <Flex alignItems={{ default: 'alignItemsCenter' }}>
+                    <Flex grow={{ default: 'grow' }} alignItems={{ default: 'alignItemsBaseline' }}>
+                        <FlexItem>
+                            <b>{_("Kernel patching")}</b>
+                        </FlexItem>
+                        <FlexItem>
+                            {state}
+                        </FlexItem>
+                    </Flex>
+                    <Flex>
+                        <Button variant="secondary"
+                                isSmall
+                                disabled={!this.props.privileged}
```

Comment:
I thought `Button` requires `isDisabled` -- the button does not actually get disabled when dropping privvs. Also, in that case shouldn't this be wrapped into a Tooltip, to explain why it is disabled?

I tried to open the page in limited access mode (or reload after dropping privs), and just get a "log: Could not determine kpatch availability.", and the button does not appear at all. That is okay I think, it's just inconsistent with how the page looks like after dropping privs. (However, not a biggie).

The "Automatic updates" option behaves correctly.

Reply:
Right, so the thing seems to be:
1. If you just drop privileges, the button is disabled
2. If you re-load page, it fails to read if needed packages are installed. Using `check_missing_packages` to also know, if these packages are even available.